eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php

```diff
@@ -854,6 +856,48 @@ public function testIndexingSpecialFullTextCases($text, $searchForText, array $i
         $this->fail('Failed to find required Content in search results');
     }
 
+    /**
+     * Check if FullText indexing works for email addresses.
+     *
+     * @dataProvider getEmailAddressesCases
+     */
+    public function testIndexingEmailFieldCases(string $email, string $searchForText)
+    {
+        $repository = $this->getRepository();
+        $searchService = $repository->getSearchService();
+
+        $content = $this->createContentEmailWithAddress($email, [2]);
+        $this->refreshSearch($repository);
+
+        $criterion = new Criterion\FullText($searchForText);
+        $query = new Query(['filter' => $criterion]);
+        $result = $searchService->findContent($query);
+
+        // for some cases there might be more than one hit, so check if proper one was found
+        foreach ($result->searchHits as $searchHit) {
+            if ($content->contentInfo->id === $searchHit->valueObject->versionInfo->contentInfo->id) {
+                return;
+            }
+        }
+        $this->fail('Failed to find required Content in search results');
```

Comment:
Could you please change $this->fail(...) to assertion?  

```php
	$found = false;
        foreach ($result->searchHits as $searchHit) {
            if ($content->contentInfo->id === $searchHit->valueObject->versionInfo->contentInfo->id) {
                $found = true;
                break;
            }
        }
        
        
        self::assertTrue($found, 'Failed to find required Content in search results');
```

Reply:
I changed this into two functions testIndexingEmailFieldCases and testIndexingSpecialFullTextCases